aten/src/ATen/cuda/CUDAApplyUtils.cuh

```diff
@@ -126,41 +133,252 @@ void rearrangeDims(detail::TensorInfo<T1, IndexType>* aInfo,
 #define AT_APPLY_THREADS_PER_BLOCK 32 * 16
 #define AT_APPLY_BLOCKS_PER_SM 4
 
+// The `step` argument is used to support Op that operates on multiple elements
```

Comment:
Could we expand this into a longer doc about what ApplyOpX does, and what all the template arguments mean?

Reply:
See the `NOTE [ CUDA_tensor_applyN helpers ]` at line 550. Let me know if I should move it up..

Reply:
Oh, I guess the Note is below...